[QUESTION] `failed to launch chromium` error while running `npx jest` within Dockerfile.bionic

**Context:**
I am using this container
https://github.com/microsoft/playwright/blob/master/docs/docker/Dockerfile.bionic


After running the container, install all dependencies  successfully, I have the following error:

**Command used to run the container:**

```
docker run -it --name playwright  -v "C:\myProject\web":/home mcr.microsoft.com/playwright bash
```

**Command used to install project dependencies:**

```
npm install 
```

Command to run tests: 

```
npx jest 
```

**Error:**

```
root@f542143afa64:/home/e2e_tests# npm run test
```

> catalogue-testing-framework-with-playwright@1.0.0 test /home/e2e_tests
> npx jest

```
Error: Failed to launch chromium because executable doesn't exist at /root/.cache/ms-playwright/chromium-827102/chrome-linux/chrome
Try re-installing playwright with "npm install playwright"
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
    at Chromium._launchProcess (/home/e2e_tests/node_modules/playwright/lib/server/browserType.js:128:19)
    at async Chromium._innerLaunch (/home/e2e_tests/node_modules/playwright/lib/server/browserType.js:70:62)
    at async ProgressController.run (/home/e2e_tests/node_modules/playwright/lib/server/progress.js:85:28)
    at async Chromium.launch (/home/e2e_tests/node_modules/playwright/lib/server/browserType.js:53:25)
    at async BrowserServerLauncherImpl.launchServer (/home/e2e_tests/node_modules/playwright/lib/browserServerImpl.js:35:25)
    at async PlaywrightRunner.launchServer (/home/e2e_tests/node_modules/jest-playwright-preset/lib/PlaywrightRunner.js:61:48)
    at async PlaywrightRunner.getTests (/home/e2e_tests/node_modules/jest-playwright-preset/lib/PlaywrightRunner.js:96:40)
    at async PlaywrightRunner.runTests (/home/e2e_tests/node_modules/jest-playwright-preset/lib/PlaywrightRunner.js:123:30)
    at async TestScheduler.scheduleTests (/home/e2e_tests/node_modules/@jest/core/build/TestScheduler.js:321:13)
    at async runJest (/home/e2e_tests/node_modules/@jest/core/build/runJest.js:376:19)
```

Any idea to what is going wrong here? thx

my package.json

```
{
  "name": "catalogue-testing-framework-with-playwright",
  "version": "1.0.0",
  "description": " E2E testing framework",
  "main": "index.js",
  "scripts": {
    "lint": "eslint test/**/*.{js,ts,tsx} --quiet --fix",
    "pretty": "npx prettier --no-config test/**/*.{js,ts,tsx} --write",
    "test": "npx jest"
  },
  "dependencies": {
    "chromedriver": "85.0.1",
    "jest": "26.6.3",
    "jest-playwright": "0.0.1",
    "jest-playwright-preset": "^1.4.0",
    "playwright": "^1.5.2",
    "ts-jest": "26.4.4",
    "typescript": "4.0.5"
  },
  "devDependencies": {
    "@types/jest": "26.0.15",
    "@types/node": "13.13.30",
    "eslint": "^6.6.0",
    "expect-playwright": "^0.2.6",
    "jest-circus": "^26.6.3",
    "jest-html-reporters": "^2.1.0",
    "playwright-core": "^1.5.2",
    "prettier": "2.1.2"
  },
  "license": "MIT",
}
```

> (sorry for the long reply - we have a huge stream of incoming requests)
> 
> > I am using this container
> > https://github.com/microsoft/playwright/blob/master/docs/docker/Dockerfile.bionic
> 
> @fasatrix did you just build it yourself and then use it? How would I reproduce this error locally?

@aslushnikov No worries I know you guys are busy..

I am using the following repo only as example but the behavior is exactly the same as for mine

Platform windows 10

Steps: 

1.  cd c:\temp (or wherever you want to)
2. git clone https://github.com/marcosvega91/msw-ssl.git
3. cd msw-ss
4. docker run -it --name playwright -v C:\Users\FSantoro\Projects\WebstormProjects\msw-ssl:/home mcr.microsoft.com/playwright bash
5. At this stage your prompt should be root
6. cd /home/src
7. npm install
8. npm run test

Error

root@11675bad909f:/home# npm run test

> msw-ssl@0.1.0 test /home
> jest

Error: Failed to launch chromium because executable doesn't exist at /root/.cache/ms-playwright/chromium-844399/chrome-linux/chrome
Try re-installing playwright with "npm install playwright"
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
    at Chromium._launchProcess (/home/node_modules/playwright/lib/server/browserType.js:134:19)
    at async Chromium._innerLaunch (/home/node_modules/playwright/lib/server/browserType.js:73:62)
    at async ProgressController.run (/home/node_modules/playwright/lib/server/progress.js:85:28)
    at async Chromium.launch (/home/node_modules/playwright/lib/server/browserType.js:55:25)
    at async BrowserServerLauncherImpl.launchServer (/home/node_modules/playwright/lib/browserServerImpl.js:35:25)
    at async PlaywrightRunner.launchServer (/home/node_modules/jest-playwright-preset/lib/PlaywrightRunner.js:62:48)
    at async PlaywrightRunner.getTests (/home/node_modules/jest-playwright-preset/lib/PlaywrightRunner.js:97:40)
    at async PlaywrightRunner.runTests (/home/node_modules/jest-playwright-preset/lib/PlaywrightRunner.js:121:30)
    at async TestScheduler.scheduleTests (/home/node_modules/@jest/core/build/TestScheduler.js:321:13)
    at async runJest (/home/node_modules/@jest/core/build/runJest.js:376:19)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! msw-ssl@0.1.0 test: `jest`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the msw-ssl@0.1.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-02-02T02_15_30_862Z-debug.log
root@11675bad909f:/home#

--- a/utils/docker/Dockerfile.bionic
+++ b/utils/docker/Dockerfile.bionic
@@ -77,16 +77,17 @@ RUN apt-get update && apt-get install -y python3.8 python3-pip && \
 
 # === BAKE BROWSERS INTO IMAGE ===
 
+ENV PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
+
+RUN mkdir /ms-playwright && chmod 777 /ms-playwright
+
 # 1. Add tip-of-tree Playwright package to install its browsers.
 #    The package should be built beforehand from tip-of-tree Playwright.
 COPY ./playwright.tar.gz /tmp/playwright.tar.gz
 
 # 2. Install playwright and then delete the installation.
-#    Browsers will remain downloaded in `/home/pwuser/.cache/ms-playwright`.
+#    Browsers will remain downloaded in `/ms-playwright`.
 RUN su pwuser -c "mkdir /tmp/pw && cd /tmp/pw && npm init -y && \
     npm i /tmp/playwright.tar.gz" && \
     rm -rf /tmp/pw && rm /tmp/playwright.tar.gz
 
-# 3. Symlink downloaded browsers for root user
-RUN mkdir /root/.cache/ && \
-    ln -s /home/pwuser/.cache/ms-playwright/ /root/.cache/ms-playwright

--- a/utils/docker/Dockerfile.focal
+++ b/utils/docker/Dockerfile.focal
@@ -76,16 +76,17 @@ RUN apt-get update && apt-get install -y python3.8 python3-pip && \
 
 # === BAKE BROWSERS INTO IMAGE ===
 
+ENV PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
+
+RUN mkdir /ms-playwright && chmod 777 /ms-playwright
+
 # 1. Add tip-of-tree Playwright package to install its browsers.
 #    The package should be built beforehand from tip-of-tree Playwright.
 COPY ./playwright.tar.gz /tmp/playwright.tar.gz
 
 # 2. Install playwright and then delete the installation.
-#    Browsers will remain downloaded in `/home/pwuser/.cache/ms-playwright`.
+#    Browsers will remain downloaded in `/ms-playwright`.
 RUN su pwuser -c "mkdir /tmp/pw && cd /tmp/pw && npm init -y && \
     npm i /tmp/playwright.tar.gz" && \
     rm -rf /tmp/pw && rm /tmp/playwright.tar.gz
 
-# 3. Symlink downloaded browsers for root user
-RUN mkdir /root/.cache/ && \
-    ln -s /home/pwuser/.cache/ms-playwright/ /root/.cache/ms-playwright

